sdk/communication/communication-common/test/public/node/entraCommunicationTokenCredential.spec.ts

```diff
@@ -27,83 +29,102 @@ const tokenCredential: TokenCredential = {
+    },
   });
 
-  it("Token exchange is called when passing Entra credential", async function () {
-    const scope = successApiMock();
-    const getTokenSpy: sinon.SinonSpy = sinon.spy(tokenCredential, "getToken");
+describe("Entra CommunicationTokenCredential", function () {
+  let getTokenSpy: sinon.SinonSpy;
 
-    const credential = new AzureCommunicationTokenCredential({
-      resourceEndpoint,
-      tokenCredential,
-      scopes,
-    });
+  beforeEach(function() {
+    getTokenSpy = sinon.spy(tokenCredential, "getToken");
+  });
 
-    const tokenResult = (await credential.getToken()).token;
-    assert.strictEqual(tokenResult, acsToken);
-    assert.isTrue(getTokenSpy.alwaysCalledWithExactly(scopes, undefined));
-    assert.isTrue(getTokenSpy.callCount > 0);
-    assert.isTrue(scope.isDone());
+  afterEach(function() {
+    nock.cleanAll();
     getTokenSpy.restore();
   });
 
-  it("Token exchange is called with default scopes when not passing scopes on AzureCommunicationTokenCredential", async function () {
-    const scope = successApiMock();
-    const defaultScopes = ["https://communication.azure.com/clients/.default"];
-    const getTokenSpy: sinon.SinonSpy = sinon.spy(tokenCredential, "getToken");
+  const testCases = [
+    {
+      successMock: communicationSuccessApiMock,
+      mock: communicationApiMock,
+      apiVersion: "2024-04-01-preview",
+      scopes: communicationClientsScope,
+      endpoint: comunicationClientsEndpoint,
+    },
+    { 
+      successMock: teamsSuccessApiMock, 
+      mock: teamsApiMock,
+      apiVersion: "2025-03-02-preview",
+      scopes: teamsExtensionScope, 
+      endpoint: teamsExtensionEndpoint 
+    },
+  ];
 
-    const entraTokenCredentialOptions: EntraCommunicationTokenCredentialOptions = {
-      resourceEndpoint: resourceEndpoint,
-      tokenCredential: tokenCredential,
-    };
+  testCases.forEach(({ successMock, mock, apiVersion, scopes, endpoint }) => {
+    it('Token exchange is called when passing Entra credential ' + apiVersion, async function () {
+      const apiMock = successMock();
 
-    const credential = new AzureCommunicationTokenCredential(entraTokenCredentialOptions);
+      const credential = new AzureCommunicationTokenCredential({
+        resourceEndpoint,
+        tokenCredential,
+        scopes: [scopes],
+      });
 
-    const tokenResult = (await credential.getToken()).token;
-    assert.strictEqual(tokenResult, acsToken);
-    assert.isTrue(getTokenSpy.alwaysCalledWithExactly(defaultScopes, undefined));
-    assert.isTrue(getTokenSpy.callCount > 0);
-    assert.isTrue(scope.isDone());
-    getTokenSpy.restore();
-  });
+      const tokenResult = (await credential.getToken()).token;
+      assert.strictEqual(tokenResult, acsToken);
+      assert.isTrue(getTokenSpy.alwaysCalledWithExactly([scopes], undefined));
+      assert.isTrue(getTokenSpy.callCount > 0);
+      assert.isTrue(apiMock.isDone());
+      getTokenSpy.restore();
```

Comment:
Can we verify if the correct endpoint is being called based on the provided scopes?